java/opendcs/src/main/java/decodes/cwms/CwmsLocationLevelDAO.java

```diff
@@ -0,0 +1,300 @@
+        if (converter == null)
+        {
+            throw new NoConversionException("No converter available from " + fromUnits + " to " + toUnits);
+        }
+        
+        return converter.convert(value);
+    }
+    
+    /**
+     * Common helper method to execute location level query using DaoHelper
+     * @param helper The DaoHelper to use for the query
+     * @param locationLevelId The location level identifier
+     * @param targetUnits The desired units for the value
+     * @return The latest LocationLevelValue or null if none found
+     * @throws SQLException on database error
+     */
+    private LocationLevelValue executeLocationLevelQuery(DaoHelper helper, String locationLevelId,
+                                                         String targetUnits) throws SQLException
+    {
+        // Build query based on whether we're filtering by source units
+        String query;
+        Object[] params;
+
+        query =
+            "SELECT CONSTANT_LEVEL AS LOCATION_LEVEL_VALUE, " +
+            "LEVEL_DATE AS LOCATION_LEVEL_DATE, LEVEL_UNIT " +
+            "FROM CWMS_20.AV_LOCATION_LEVEL " +
+            "WHERE LOCATION_LEVEL_ID = ? AND UNIT_SYSTEM = 'SI' " +
+            "ORDER BY LEVEL_DATE DESC " +
+            "FETCH FIRST 1 ROWS ONLY";
+        params = new Object[] { locationLevelId };
+
+        final String[] sourceUnits = new String[] { null };
+        
+        return helper.getSingleResultOr(query, rs -> {
+            LocationLevelValue value = new LocationLevelValue();
+            double levelValue = rs.getDouble("LOCATION_LEVEL_VALUE");
+            String dbUnits = rs.getString("LEVEL_UNIT");
+
+            sourceUnits[0] = dbUnits;
+            
+            // Perform unit conversion if needed
+            if (targetUnits != null && !targetUnits.isEmpty() && 
+                sourceUnits[0] != null && !sourceUnits[0].equalsIgnoreCase(targetUnits))
+            {
+                try
+                {
+                    levelValue = convertUnits(levelValue, sourceUnits[0], targetUnits);
+                    value.setUnits(targetUnits);
+                }
+                catch (Exception ex)
+                {
+                    log.warn("Failed to convert units from {} to {}: {}",
+                            sourceUnits[0], targetUnits, ex.getMessage());
+                    value.setUnits(sourceUnits[0]);
+                }
+            }
+            else
+            {
+                value.setUnits(sourceUnits[0]);
+            }
+            
+            value.setLevelValue(levelValue);
+            value.setLevelDate(rs.getTimestamp("LOCATION_LEVEL_DATE"));
+            value.setLocationLevelId(locationLevelId);
+            value.setQualityCode(0);
+            return value;
+        }, null, params);
+    }
+    
+    
+    /**
+     * Close the DAO and free resources
+     */
+    @Override
+    public void close()
+    {
+        // Nothing to close in stateless implementation
```

Comment:
throw a runtime except to make sure this never gets called until we can ditch DaoBase